src/main/java/ru/yandex/clickhouse/response/ByteFragmentUtils.java

```diff
@@ -181,7 +193,29 @@ public static Object parseArray(ByteFragment value, Class elementClass) {
                 } else if (elementClass == BigInteger.class){
                     BigInteger bigIntegerValue = new BigInteger(fragment.asString(true));
                     java.lang.reflect.Array.set(array, index++, bigIntegerValue);
-                } else {
+                } else if (elementClass == Float.class) {
+                    Float floatValue = Float.parseFloat(fragment.asString());
+                    java.lang.reflect.Array.set(array, index++, floatValue);
+                } else if (elementClass == Double.class) {
+                    Double doubleValue = Double.parseDouble(fragment.asString());
+                    java.lang.reflect.Array.set(array, index++, doubleValue);
+                } else if (elementClass == Date.class) {
+                    Date dateValue;
+                    try {
+                        dateValue = new Date(DATE_FORMAT.parse(fragment.asString()).getTime());
```

Comment:
Timezone is a property of ClickhouseConnection. It is initialized on ClickhouseConnection initialization either by requesting it from server or from ClickHouseProperties (depending on properties).
The code is here ru.yandex.clickhouse.ClickHouseConnectionImpl#initTimeZone

Timezone is also propagated to ClickHouseResultSet by setting it to local SimpleDateFormat's.
ru.yandex.clickhouse.response.ClickHouseResultSet#initTimeZone

I guess you should somehow pass these DateFormat's to the parsing methods.